File: src/main/java/org/example/model/ProbeResult.java
package org.example.model;

import java.time.LocalDateTime;

public class ProbeResult {

    private LocalDateTime timestamp;
    private double latencyMs;
    private boolean reachable;
    private boolean success;

    public ProbeResult() {
    }

    public ProbeResult(LocalDateTime timestamp,
                       double latencyMs,
                       boolean reachable,
                       boolean success) {
        this.timestamp = timestamp;
        this.latencyMs = latencyMs;
        this.reachable = reachable;
        this.success = success;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    public double getLatencyMs() {
        return latencyMs;
    }

    public void setLatencyMs(double latencyMs) {
        this.latencyMs = latencyMs;
    }

    public boolean isReachable() {
        return reachable;
    }

    public void setReachable(boolean reachable) {
        this.reachable = reachable;
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    @Override
    public String toString() {
        return "ProbeResult{" +
                "timestamp=" + timestamp +
                ", latencyMs=" + latencyMs +
                ", reachable=" + reachable +
                ", success=" + success +
                '}';
    }
}

File: node-version/src/model/ProbeResult.js
class ProbeResult {

    constructor(
        timestamp,
        latencyMs,
        reachable,
        success
    ) {
        this.timestamp = timestamp;
        this.latencyMs = latencyMs;
        this.reachable = reachable;
        this.success = success;
    }

    toString() {
        return JSON.stringify({
            timestamp: this.timestamp,
            latencyMs: this.latencyMs,
            reachable: this.reachable,
            success: this.success
        });
    }
}

export default ProbeResult;